core/commands/partials/dedicated_configuration_generator.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+
+require_relative '../../services/configuration_generator'
+require_relative '../../models/configuration'
+
+# The class generates the MDBCI configuration for computers that have been setup beforehand
+class DedicatedConfigurationGenerator < BaseCommand
+  KEY_FILE_NAME_SUFFIX = 'public_key'
+
+  def execute(name)
+    setup_result = setup_command(name)
+    return generate_result if setup_result.error?
+
+    check_nodes_boxes_and_setup_provider
+    @ui.info("Nodes provider = #{@provider}")
+    generate_result = generate
+    return generate_result if generate_result.error?
+
+    generate_configuration_info_files
+    SUCCESS_RESULT
+  end
+
+  def setup_command(name)
+    return Result.error('Please specify destination') if name.nil?
+
+    @boxes = @env.box_definitions
+    @configuration_generator = ConfigurationGenerator.new(@ui, @env)
+    @configuration_path = File.absolute_path(name.to_s)
+    begin
+      @config = ConfigurationTemplate.new(
+        File.expand_path(@env.template_file), @env.box_definitions
+      ).node_configurations
```

Comment:
You must store reference to the ConfigurationTemplate class, not to the array of Hashes.